python/taichi/misc/np2ply.py

```diff
@@ -128,6 +125,19 @@ def add_vertex_rgba(self, r: np.array, g: np.array, b: np.array, a: np.array):
         self.add_vertex_channel("blue", "float", b)
         self.add_vertex_channel("Alpha", "float", a)
 
+    # pass ti vector/matrix tensor directly
+    def add_vertex_rgba(self, rgba):
+        if not isinstance(rgba, np.ndarray):
+            rgba = rgba.to_numpy()
+        channels = rgba.shape[rgba.ndim-1]
+        assert channels == 4, "The dimension for rgba should be 4"
+        n = rgba.size // channels

```

Comment:
I think self.num_vertex is better here.

Reply:
This makes sense that `n` must equal to `self.num_vertex`. Nonetheless, I would prefer to calculate `n` then add an assertion here. Because people may pass a `ti.vector` tensor with the wrong size without realizing this, so we should inform them here.

What would you suggest?

Reply:
Yeah, I was worries about the same thing, I think that's exactly we don't want to calculate n before we do the assert

Reply:
I think our disagreement here is somewhat the same as the next conversation, I will address it there. Thanks